### DISCUSSION TITLE
Redundancy and Security Onion

### DISCUSSION THREAD

**Discussion by HundleBun** [Security-Onion-Solutions/securityonion#6414] [Unsupported Versions] [answered]
I've been playing around with SO in a virtual environment at home and recently spoke with our Network Admin and Server Admin about the tool as we don't really have a logging/monitoring solution in place at work. One of the first things that was asked was in regards to redundancy.

We run active/active active/passive for most of our hardware and they are concerned with redundancy/high-availability. Can SO be setup in a way that we could have redundancy should the main server hosting it goes down? What might this look like? Is there anything in documentation I can look up and/or point them to?

Any help is appreciated!

**Comment by dougburks** [comment, marked as answer]
You can do data replication using Elastic clustering:
https://docs.securityonion.net/en/2.3/elasticsearch.html#elastic-clustering

However, please keep in mind that there is no full redundancy for the manager node at this time. We may consider that in the future.

**Comment by hacker0ni** [comment]
We were worried about the same stuff, to backup the customizations/configurations we've made to the grid, we're backing up `/opt/so/saltstack/local` to a Git repo and making the changes there (then pulling it from repo etc.). This is still not the best way to backup the system, because in the event of the manager machine going down, we'd still have to redeploy the grid. However it helps with config restoration which is the biggest difference between the official repo and your grid.
